exercises/concept/story-name/.meta/design.md

```diff
@@ -0,0 +1,28 @@
+# Design
+
+## Goal
+
+The goal of this exercise is to teach students how to consume and make generic code.
+
+## Learning objectives
+
+- Understand what a type variable is.
+- Know how to define generic types.
+- Know how to define generic functions.
+- Introduce higher order functions.
+
+## Out of scope
+
+- Covariance and contravariance.
+
```

Comment:
Maybe add Never, Phantom Types and Opaque types as out of scope, although it would probably be useful to mention them in passing so that students know they exist

Reply:
Well for Never, I'm not sure we even need to mention it since it's not really a generic, just a standard type that you have no way to construct. Phantom types, I'm not sure they are out of scope, at least introducing them. That will probably depend on if the concept grows a bit big already without them. As for opaque types, they are not linked to generics in any specific way, they are just normal types. The only thing special is that you don't export the types internal so it's more of public/private and module-related concept.

Reply:
Yep, that is true, although in my head they are all in the "advanced types" bucket :)